
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=547734
iptables-apply uses /etc/network/iptables as default rulefile and so ip6tables-apply does


Index: b/iptables-apply
===================================================================
--- a/iptables-apply	2010-05-21 06:48:13.000000000 -0400
+++ b/iptables-apply	2010-06-12 13:25:20.321045229 -0400
@@ -11,7 +11,6 @@
 VERSION=1.0
 
 TIMEOUT=10
-DEFAULT_FILE=/etc/network/iptables
 
 function blurb()
 {
@@ -87,6 +86,19 @@
 	shift
 done
 
+case "$PROGNAME" in
+	(*6*)
+		SAVE=ip6tables-save
+		RESTORE=ip6tables-restore
+		DEFAULT_FILE=/etc/network/ip6tables
+		;;
+	(*)
+		SAVE=iptables-save
+		RESTORE=iptables-restore
+		DEFAULT_FILE=/etc/network/iptables
+		;;
+esac
+
 FILE="${1:-$DEFAULT_FILE}";
 
 if [[ -z "$FILE" ]]; then
@@ -99,17 +111,6 @@
 	exit 2
 fi
 
-case "${0##*/}" in
-	(*6*)
-		SAVE=ip6tables-save
-		RESTORE=ip6tables-restore
-		;;
-	(*)
-		SAVE=iptables-save
-		RESTORE=iptables-restore
-		;;
-esac
-
 COMMANDS=(tempfile "$SAVE" "$RESTORE")
 
 for cmd in "${COMMANDS[@]}"; do
